server/src/Service/InteractionService.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace App\Service;
+
+use App\Dao\RevisionDao;
+
+class InteractionService {
+
+	/**
+	 * @var RevisionDao
+	 */
+	private $revisionDao;
+
+	/**
+	 * @param RevisionDao $revisionDao
+	 */
+	public function __construct( RevisionDao $revisionDao ) {
+		$this->revisionDao = $revisionDao;
+	}
+
+	/**
+	 * @param string[] $users
+	 * @param int|null $startDate
+	 * @param int|null $endDate
+	 * @param int $limit
+	 * @param string|null $continue
+	 * @return array
+	 */
+	public function getInteraction(
+		$users, $startDate = null, $endDate = null, $limit = 50, $continue = null
+	) {
+		$this->validateUsers( $users );
+		$this->validateLimit( $limit );
+
+		list( $revisionIds, $continue ) = $this->revisionDao->getUserRevisionsInCommonPages(
+				$users,  $startDate, $endDate, $limit, $continue
+		);
+
+		$revisions = [];
+		if ( $revisionIds ) {
+			$revisions = $this->revisionDao->getRevisionDetails( $revisionIds );
+			$revisions = $this->parseAndMapRevisions( $revisions );
+		}
+
+		return [ $revisions, $continue ];
+	}
+
+	/**
+	 * @param int[] $revisions
+	 * @return array
+	 */
+	private function parseAndMapRevisions( $revisions ) {
+		$interaction = [];
+		foreach ( $revisions as $revision ) {
```

Comment:
nit-pick.. you could use [array_map()](http://php.net/manual/en/function.array-map.php) with an anonymous callback instead of the foreach.